AsyncDisplayKit/Private/_ASHierarchyChangeSet.m

```diff
@@ -301,6 +333,28 @@ - (instancetype)initWithChangeType:(_ASHierarchyChangeType)changeType indexPaths
   return self;
 }
 
+// Create a mapping out of changes indexPaths to a {@section : [indexSet]} fashion
+// e.g. changes: (0 - 0), (0 - 1), (2 - 5)
+//  will become: {@0 : [0, 1], @2 : [5]}
++ (NSDictionary *)sectionToIndexSetMapFromChanges:(NSArray *)changes ofType:(_ASHierarchyChangeType)changeType
+{
+  NSMutableDictionary *sectionToIndexSetMap = [NSMutableDictionary dictionary];
+  for (_ASHierarchyItemChange *change in changes) {
+    NSAssert(change.changeType == changeType, @"The map we created must all be of the same changeType as of now");
+    for (NSIndexPath *indexPath in change.indexPaths) {
+      NSString *sectionKey = @(indexPath.section).stringValue;
+      NSMutableIndexSet *indexSet = sectionToIndexSetMap[sectionKey];
```

Comment:
Why parse the section number into a string? Why not use the NSNumber as the key?